### PR TITLE
Web API to serve `samlidp gcp-workforce` oneoff script

### PR DESCRIPTION
**Pull request by flyinghermit**
This PR adds web API that returns a oneoff script configured to run `teleport integration configure samlidp gcp-workforce` command.

### CODE REVIEW

**Comment by marcoandredinis** [comment]
I don't think we should create more shell scripts.
They tend to be much less testable and harder to evolve.

Can we please use the `oneoff` package (`oneoff.BuildScript`)?

**Comment by flyinghermit** [comment]
> I don't think we should create more shell scripts.
> They tend to be much less testable and harder to evolve.
> 
> Can we please use the oneoff package (oneoff.BuildScript)?

I completely agree with the "less testable and harder to evolve". Though is `oneoff` the right script for this one? `oneoff` looks like is specifically to install Teleport but the script introduced in this PR does not involve installing Teleport at all, merely calling curl and `gcloud` command.

**Comment by marcoandredinis** [comment]
It downloads `teleport` binary and runs oneoff command, like `./teleport configure ...`

**Comment by flyinghermit** [comment]
Taking this PR to draft now that the script is to be replaced with GCP IAM client that is added in the PR https://github.com/gravitational/teleport/pull/39820. 

I will be updating this PR to just spit out the oneoff script.

**Comment by flyinghermit** [comment]
@marcoandredinis @r0mant I've reopened this PR for review. It just now serves the oneoff script as the heavy lifting went in with https://github.com/gravitational/teleport/pull/39820

File: lib/web/integrations_samlidp.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	argsList := []string{
+		"integration", "configure", "samlidp", "gcp-workforce",
+		fmt.Sprintf("--org-id=%s", libutils.UnixShellQuote(queryParams.Get("orgId"))),
+		fmt.Sprintf("--pool-name=%s", libutils.UnixShellQuote(queryParams.Get("poolName"))),
+		fmt.Sprintf("--pool-provider-name=%s", libutils.UnixShellQuote(queryParams.Get("poolProviderName"))),
+		fmt.Sprintf("--idp-metadata-url=%s", libutils.UnixShellQuote(samlIdPMetadataURL)),
+	}
+	script, err := oneoff.BuildScript(oneoff.OneOffScriptParams{
+		TeleportArgs:   strings.Join(argsList, " "),
+		SuccessMessage: "Success! You can now go back to the browser to complete the database enrollment.",
```
Comment (r0mant) [inline lib/web/integrations_samlidp.go:60]: This should probably something other than "database enrollment".

**Review by r0mant — APPROVED** [review]
lgtm - have you tested the full end-to-end flow with this?

File: lib/web/integrations_samlidp.go (hunk 1 above)
Comment (flyinghermit) [reply lib/web/integrations_samlidp.go]: Uh Oh! updated in https://github.com/gravitational/teleport/pull/39714/commits/cb64da8b839292dda925a36b514c985ea1c9d5a9

**Comment by flyinghermit** [comment]
> have you tested the full end-to-end flow with this?

I have manually tested the script returned from the API to run it with teleport binary of my choice. I am trying to get the dev release so I can have a proper end-to-end test. I wont merge this PR until that is done.

**Review by marcoandredinis — APPROVED** [review]

**Comment by flyinghermit** [comment]
While I am not able to get the dev release published on the CDN, the pipeline did publish in the `get.gravitational.com` ([tarball link](https://get.gravitational.com/teleport-v16.0.0-dev.flyinghermit.4-linux-amd64-bin.tar.gz)). I manually tested with the teleport binary from that tarball and it worked. I am going to merge this PR now so it gets in on time.

**Pull request merged**